include/twin_private.h

```diff
@@ -125,6 +138,11 @@ typedef int8_t twin_gfixed_t;
 #define twin_dfixed_div(a, b) \
     ((((twin_dfixed_t) (a)) << 8) / ((twin_dfixed_t) (b)))
 
+#define twin_xfixed_mul(a, b) \
+    (twin_xfixed_t)((((__int128_t) (a)) * ((__int128_t) (b))) >> 32)
```

Comment:
I am curious about the reason of using `__int128_t`, because the function primarily involves  `twin_xfixed_to_fixed()` and `twin_fixed_to_xfixed()`, and `twin_fixed_t` is defined as a 16.16 fixed-point format.

I understand the necessity of using `__int128_t` in `twin_xfixed_mul(p_ry_divided_rx_x, px_x)` (line 311 in path.c), because `p_ry_divided_rx_x` has already undergone one twin_xfixed_t operation:
```c
twin_xfixed_t p_ry_divided_rx_x = twin_xfixed_div(ry_x, rx_x);
```
This means that operand in `twin_xfixed_mul(p_ry_divided_rx_x, px_x)` actually undergo two `twin_xfixed_t` operations when `twin_xfixed_mul()` is executed.

However, why not convert `p_ry_divided_rx_x` to `twin_fixed_t` in advance, as done in line 296 in path.c ?
```c
twin_fixed_t L = twin_xfixed_to_fixed(twin_xfixed_div(px_x, prx_x) +
                                          twin_xfixed_div(py_x, pry_x));
```
If `p_ry_divided_rx_x` were converted to `twin_fixed_t` (16.16 format) beforehand, it would ensure that all operands in `twin_dfixed_div()` and `twin_xfixed_mul()` undergo their first `twin_xfixed_t` operation when these functions are executed.

Reply:
I think you misunderstood it, we didn't perform fixed-point type conversion in `twin_xfixed_mul` and `twin_xfixed_div`. Converting `a` and `b` to `__int128_t` is to avoid calculation overflow. Similar to  how we handle `twin_fixed_mul` or `twin_fixed_div`
```
#define twin_fixed_mul(a, b) ((twin_fixed_t) (((int64_t) (a) * (b)) >> 16))
#define twin_fixed_div(a, b) ((twin_fixed_t) ((((int64_t) (a)) << 16) / (b)))
```

> However, why not convert p_ry_divided_rx_x to twin_fixed_t in advance, as done in line 296 in path.c ?

I didn't get it, what's the relation between `p_ry_divided_rx_x ` and this formula.

Reply:
Yes, I misunderstand. Therefore, when performing multiplication or division, it is necessary to use a data type whose size is twice as large as the original operand's data type to avoid overflow.

Reply:
```c
#define twin_sfixed_mul(a, b) \
    ((((twin_sfixed_t) (a)) * ((twin_sfixed_t) (b))) >> 4)
#define twin_sfixed_div(a, b) \
    ((((twin_sfixed_t) (a)) << 4) / ((twin_sfixed_t) (b)))

#define twin_dfixed_mul(a, b) \
    ((((twin_dfixed_t) (a)) * ((twin_dfixed_t) (b))) >> 8)
#define twin_dfixed_div(a, b) \
    ((((twin_dfixed_t) (a)) << 8) / ((twin_dfixed_t) (b)))
```
Therefore, this part should also use a data type that is twice the size of the original operand's data type.

Reply:
`__int128_t` is a GNU extension. You should add `FIXME` note for portable solutions.